### DISCUSSION TITLE
Extendibility

### DISCUSSION THREAD

**Discussion by RegisWeduLabs** [simylein/fluxify#3] [General]
Currently, if you want to keep a fluxify project updated with this repo, it can cause quite a few issues if you modify anything...  I think it would be nice to change the design a bit so we can extend the core functionnality without loosing the benifits of updating the package.  A good example would be to create a custom data source (api, postgres, etc...).  Or extending functionnalities (Docs, cache(new), events, logging, etc...) without modifying the actual core packages.  Toughts?

**Comment by simylein** [comment]
Currently, you can just replace the lib folder to update because everything in the src folder is your code and lib is the library. 
As for including databases like postgres you would need to add your first dependency to this dependency free project which you are of course welcome to do, it's just not something I plan to provide. If bun starts shipping a mysql or postgres client in the future we could implement this. As for extending you have raw access to request and response cycle which you can use for pretty much anything bun is able to do. File upload and streaming, non json responses, etc... just some of my thoughts.

**Comment by RegisWeduLabs** [comment]
I think I was more talking about extending existing library functionnalities.  Say I want to add custom data in the docs, like the schemas.  Currently I can either modify the file, or copy the function in another file and do what I have to - but then we loose the update benefits.  I think I'll simply give you an example of what I mean by that then we can take it from there.  I'll go ahead and create a POC PR and we can discuss a bit further?

**Comment by simylein** [comment]
For schema documentation I was thinking about searching the src folder for files like `*.dto.ts` and then take the exported object and if it's a `RecursiveParser` add it to the schemas of the docs function. This would work in similar ways as the entities because you can also name them `*.entity.ts` and they will get picked up by the `bun schema:*` scripts.

**Comment by simylein** [comment]
I played around with schemas in the docs a bit and discovered that it is quite hard to do as my current implementation does not allow easy reading of schemas without executing the validation logic. I think I have to rework how `lib/validation` works for this feature. I am however looking forward to your pr because maybe you have some better ideas than me.
